bundles/org.openhab.binding.pushover/src/main/java/org/openhab/binding/pushover/internal/connection/PushoverMessageBuilder.java

```diff
@@ -0,0 +1,267 @@
+            } else {
+                // TODO [A-Za-z0-9_-]
+                // TODO Messages may be addressed to multiple specific devices by joining them with a comma (such as
+                // device=iphone,nexus5)
+                body.addFieldPart(MESSAGE_KEY_DEVICE, new StringContentProvider(device), null);
+            }
+        }
+
+        if (priority != DEFAULT_PRIORITY) {
+            if (VALID_PRIORITY_LIST.contains(priority)) {
+                body.addFieldPart(MESSAGE_KEY_PRIORITY, new StringContentProvider(String.valueOf(priority)), null);
+
+                if (priority == EMERGENCY_PRIORITY) {
+                    if (retry < MIN_RETRY_SECONDS) {
+                        logger.warn("Retry value of {} is too small. Using default value of {}.", retry,
+                                MIN_RETRY_SECONDS);
+                        body.addFieldPart(MESSAGE_KEY_RETRY,
+                                new StringContentProvider(String.valueOf(MIN_RETRY_SECONDS)), null);
+                    } else {
+                        body.addFieldPart(MESSAGE_KEY_RETRY, new StringContentProvider(String.valueOf(retry)), null);
+                    }
+
+                    if (0 < expire && expire <= MAX_EXPIRE_SECONDS) {
+                        body.addFieldPart(MESSAGE_KEY_EXPIRE, new StringContentProvider(String.valueOf(expire)), null);
+                    } else {
+                        logger.warn("Expire value of {} is invalid. Using default value of {}.", expire,
+                                MAX_EXPIRE_SECONDS);
+                        body.addFieldPart(MESSAGE_KEY_EXPIRE,
+                                new StringContentProvider(String.valueOf(MAX_EXPIRE_SECONDS)), null);
+                    }
+                }
+            } else {
+                logger.warn("Invalid 'priority', skipping. Expected: {}. Got: {}.",
+                        VALID_PRIORITY_LIST.stream().map(i -> i.toString()).collect(Collectors.joining(",")), priority);
+            }
+        }
+
+        if (url != null) {
+            if (url.length() > MAX_URL_LENGTH) {
+                throw new IllegalArgumentException(String
+                        .format("Skip sending the message as 'url' is longer than %d characters.", MAX_URL_LENGTH));
+            }
+            body.addFieldPart(MESSAGE_KEY_URL, new StringContentProvider(url), null);
+
+            if (urlTitle != null) {
+                if (urlTitle.length() > MAX_URL_TITLE_LENGTH) {
+                    throw new IllegalArgumentException(
+                            String.format("Skip sending the message as 'urlTitle' is longer than %d characters.",
+                                    MAX_URL_TITLE_LENGTH));
+                }
+                body.addFieldPart(MESSAGE_KEY_URL_TITLE, new StringContentProvider(urlTitle), null);
+            }
+        }
+
+        if (sound != null) {
+            // TODO validate sound
```

Comment:
can this be done?

Reply:
Not needed anymore. Sounds are now provided by a `ConfigOptionsProvider`.